client/src/app/videos/+video-watch/video-watch.component.ts

```diff
@@ -435,15 +436,24 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
 
       this.player.one('ended', () => {
         if (this.playlist) {
-          this.zone.run(() => this.videoWatchPlaylist.navigateToNextPlaylistVideo())
-        } else if (this.user && this.user.autoPlayNextVideo) {
+          if (
+            this.user && this.user.autoPlayNextVideoPlaylist ||
```

Comment:
put this condition in a dedicated method `isPlaylistAutoPlayEnabled()` for example